ragstack/colbertbase/colbertbase/token_embedding.py

```diff
@@ -0,0 +1,114 @@
+#
+# this is a base class for ColBERT per token based embedding
+
+from abc import ABC, abstractmethod
+from typing import List
+import uuid
+
+class PerTokenEmbeddings():
+
+    __embeddings: List[float]
+
+    def __init__(
+            self,
+            id: int,
+            part: int,
+            parent_id: uuid.UUID = None,
+            title: str = "",
+        ):
+        self.id = id
+        self.parent_id = parent_id
+        self.__embeddings = []
+        self.title = title
+        self.part =part
+
+    def add_embeddings(self, embeddings: List[float]):
+        self.__embeddings = embeddings
+
+    def get_embeddings(self) -> List[float]:
+        return self.__embeddings
+
+    def id(self):
+        return self.id
+
+    def parent_id(self):
+        return self.parent_id
+
+    def part(self):
+        return self.part
+
+class PassageEmbeddings():
+    __token_embeddings: List[PerTokenEmbeddings]
+    __text: str
+    __title: str
+    __id: uuid.UUID
+
+    def __init__(
+            self,
+            text: str,
+            title: str = "",
+            part: int = 0,
+            id: uuid.UUID = None,
```

Comment:
what is difference between `title` and `id`?

Reply:
Title should be supplied by the user or specified as metadata.
Id is to track the parent of each per token embeddings which is the chunk.
I don't use Id at the moment. So it can be deleted. I have not removed them yet.